piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -298,13 +440,80 @@ private Description updateCode(
 
     if (update) {
       Description.Builder builder = buildDescription(tree);
-      builder.addFix(SuggestedFix.replace(expr, replacementString));
+      SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+      fixBuilder.replace(expr, replacementString);
+      decrementAllSymbolUsages(expr, state, fixBuilder);
+      builder.addFix(fixBuilder.build());
       endPos = state.getEndPosition(expr);
       return builder.build();
     }
     return Description.NO_MATCH;
   }
 
+  private boolean isTreatmentGroupEnum(Symbol.ClassSymbol enumSym) {
+    // Filter out some generic names, like CONTROL to make sure we don't match the wrong
+    // TreatmentGroup
+    if (COMMON_GROUP_NAMES.contains(treatmentGroup)) {
+      return false;
+    }
+    for (Symbol fsym : enumSym.getEnclosedElements()) {
+      if (fsym.getKind().equals(ElementKind.ENUM_CONSTANT)
+          && fsym.name.toString().toLowerCase().equals(treatmentGroup)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public Description matchClass(ClassTree classTree, VisitorState visitorState) {
```

Comment:
this will work well internally, not sure how this is appropriate for external use. We need to think of a better flow here.

Reply:
I think I agree, although I am also not sure the external flow needs to be too different (except for using gradle instead of buck and integrations with whatever the users' CI is). For now, I think moving the development to the open can be done first and then we can improve the UX for external adopters :)